features/topology-map/plugins/org.opennms.features.topology.plugins.topo.asset/src/main/java/org/opennms/features/topology/plugins/topo/asset/AssetGraphDefinitionRepository.java

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2017-2017 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2017 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.features.topology.plugins.topo.asset;
+
+import java.util.Map;
+
+public interface AssetGraphDefinitionRepository {
+
+	/**
+	 * Returns Asset topology definition for given providerId
+	 * @param providerId
+	 * @return GeneratorConfig containing Asset topology definition or null if config does not exist
+	 */
+	public abstract GeneratorConfig getConfigDefinition(String providerId);
```

Comment:
This is an interface definition, therefore `public abstract` is unnecessary information, as all interfaces must overwrite these methods anyways. `public abstract` should be removed.